src/educational-technology.rst

```diff
@@ -1,18 +1,183 @@
+  demonstrate instances in which code fails, 3) explain behavior of code with visual execution, and 4) help students understand the relationship
+  between the cause of an error and its symptoms.
+
 Tutoring Systems
 ================
 
+Large class sizes and the growing number of people learning computer programming through online courses makes the 
+development of tutoring systems valuable. Tutoring systems provide visualizations and walkthroughs of the execution
+of code. Additionally, they may guide students through developing a program. Students may remain more engaged with the system
+and learn more if there is interaction that demonstrates whether or not the student is understanding and following along.
+
+.. container:: bib-item
+
+  .. bibliography:: educational-technology.bib
+    :filter: key == 'guo2013online'
+
+  Python Tutor is an online tutoring system. This is a form of program visualization that shows the state of memory (stack frames and the heap)
+  as a piece of code executes, essentially creating visual code traces. 
+
+.. todo::
+    Add information about systems that introduce the programming language a bit at a time (like SP/k)
+
+.. todo::
+    Find interactive tutoring systems
+
+Educator Support
+================
+
+In the space of education technology, there is needed support for educators. This can either be in an informal context of peer-to-peer knowledge-sharing
+or in the context of a more formal classroom setting. 
+
+Informal knowlege sharing allows peers to share information they have learned with one another. Different environments support this to 
+varying levels with some allowing users to explore and modify projects made by other users and to ask clarifying questions.
+For a more formal classroom setting, teachers can often use support in getting a better understanding of what topics
+their students are excelling at understanding and what topics their students are struggling to understand.
+
+.. container:: bib-item
+
+  .. bibliography:: educational-technology.bib
+    :filter: key == 'glassman2015overcode'
+
+  OverCode is a system designed to allow instructors of large programming classes to automatically group solutions that may have 
+  irrelavent syntactic differences but are semantically equivalent. This should allow instructors to get a high-level overview of
+  the understandings and misunderstandings of their students more quickly than filtering through raw solutions. When evaluated against
+  only having access to raw solutions, users were able to review more students' code in less time and felt that they had a better high-level
+  overview of ways that the coding problems were solved. 
+
+.. container:: bib-item
+
+  .. bibliography:: educational-technology.bib
+    :filter: key == 'head2018interactive'
+
+  CodeScoop is an interactive example extraction system that allows users to quickly pull out examples from an existing code
```

Comment:
You should justify why this is under "Educator Support" (e.g. by mentioning that the focus of the paper is on answering questions on Stack Overflow)